Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -404,107 +354,67 @@ void CGPathAddArc(CGMutablePathRef path,
     }
 
     // Call the internal function for breaking the arcs down into smaller segments
-    _CGPathAddArc(path, m, x, y, radius, startAngle, endAngle, clockwise);
+    _CGPathAddArc(path, transform, x, y, radius, startAngle, endAngle, clockwise);
 }
 
 /**
  @Status Interoperable
 */
-void CGPathMoveToPoint(CGMutablePathRef path, const CGAffineTransform* m, float x, float y) {
-    if (path == NULL) {
-        return;
-    }
-
-    if (m) {
-        CGPoint pt;
+void CGPathMoveToPoint(CGMutablePathRef path, const CGAffineTransform* transform, CGFloat x, CGFloat y) {
```

Comment:
I can't find where you are moving the current point in the geometry.  Ex: AddLine just takes the target point, not the starting point.